set.mm

```diff
@@ -532760,6 +532537,78 @@ dependency is expressed in our hypothesis (called implicit
       CDBJABCDKL $.
   $}
 
+  $( ~ spsbbi applied twice.  (Contributed by Wolf Lammen, 5-Aug-2023.) $)
+  wl-2spsbbi $p |- ( A. a A. b ( ph <-> ps ) ->
+       ( [ y / b ] [ x / a ] ph <-> [ y / b ] [ x / a ] ps ) ) $=
+    ( wb wal wsb alcom nfa1 sp sbbid sps sylbi ) ABGZFHEHPEHZFHZAECIZFDIBECIZFD
+    IGPEFJRSTFDQFKQSTGFQABECPEKPELMNMO $.
+
+  ${
+    $d a b x $.  $d a b y $.  $d a b ph $.
+    $( Infer one criterium for interchangeability from another.  (Contributed
+       by Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich $p |- ( A. a A. b ( [ a / x ] [ b / y ] ph
+                              <-> [ b / x ] [ a / y ] ph )
+        -> ( ph <-> [ x / a ] [ y / x ] [ a / y ] ph ) ) $=
+      ( wsb wb wal wl-2spsbbi sbid2vw sbbii bitri sbco4 3bitr3g ) ACEFZBDFZACDF
+      ZBEFZGEHDHPDBFZECFZRDBFECFAQBCFDBFPRBCDEITOECFASOECODBJKAECJLABCDDEMN $.
+  $}
+
+  ${
+    $d b x $.
+    $( Lemma.  Simplify substitution expression when two particular variables
+       are the same.  (Contributed by Wolf Lammen, 5-Aug-2023.) $)
+    wl-ich-lem1 $p |- ( A. x x = y ->
+                         ( [ a / x ] [ b / y ] ph <-> [ b / x ] ph ) ) $=
```

Comment:
`[ a / x ] [ b / y ] ph <-> [ b / y ] ph` seems more natural: "When substituting successively for two always equal variables, the second substitution has no effect." I think this result is interesting in its own, and not only as a lemma.

Also: the use of ~sbbid in your proof makes me think that the proof would be more direct with antecedent `A. y y = x`, or, renaming variables:
```
|- ( A. x x = y -> ( [ b / y ] [ a / x ] ph <-> [ a / x ] ph ) )
```

Reply:
I am pretty sure there is headroom for optimization.  I simply have not unlimited time.  If this is moved to main, anyone is free to improve on these theorems

Reply:
There is an extension under way.